### PR TITLE
# Task 5

### PR DESCRIPTION
**Pull request by vmakhanch**
# Task 5

Link to swagger http://music-store-backend-static-data-rs-scool-aws.s3-website-us-east-1.amazonaws.com/import-service.html

Link to CloudFront website - https://d16amwjd31tx37.cloudfront.net

## Task 5.1
Get signed url: GET https://u0a8ks9k45.execute-api.us-east-1.amazonaws.com/dev/import?name={fileName}

## Task 5.2
Done

##Additional tasks done:
- Async/await is used in lambda functions
- importProductsFile lambda is covered by unit tests ((for JS only) aws-sdk-mock can be used to mock S3 methods
- At the end of the stream the lambda function should move the file from the uploaded folder into the parsed folder (move the file means that file should be copied into parsed folder, and then deleted from uploaded folder)

### CODE REVIEW

File: import-service/src/handlers/importFileParser/importFileParser.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import parseCsv from 'csv-parser'
+
+import {ResponseStringCodes, ResponseStatusCodes} from '../../common/types'
+import {AwsGatewayAPIResponse} from '../../common/entities/AwsGatewayAPIResponse'
+import {BUCKET_NAME} from '../../common/constants'
+
+export const handler = async (event: S3Event) => {
+    try {
+        const s3 = new AWS.S3({
+            region: 'us-east-1'
```
Comment (ambroseus) [inline import-service/src/handlers/importFileParser/importFileParser.ts:12]: you can use `process.env.REGION`

**Review by ambroseus — APPROVED** [review]

File: import-service/src/handlers/importFileParser/importFileParser.ts (hunk 1 above)
Comment (vmakhanch) [reply import-service/src/handlers/importFileParser/importFileParser.ts:12]: Got it. Thanks

**Comment by ilyababich** [comment]
Got an error when trying to upload file 
image

**Comment by vmakhanch** [comment]
> Got an error when trying to upload file image

Yes. It is fine. It is because I have completed task 7 and now you should provide credentials to load a file.

**Pull request merged**